pkg/services/ngalert/models/alertmanager.go

```diff
@@ -14,6 +14,19 @@ type AlertConfiguration struct {
 	OrgID                     int64 `xorm:"org_id"`
 }
 
+// HistoricAlertConfiguration represents a previously used alerting configuration.
+type HistoricAlertConfiguration struct {
```

Comment:
I would define this WRT to `AlertConfiguration`. Something like:

```
// HistoricAlertConfiguration represents a previously used alerting configuration.
type HistoricAlertConfiguration struct {
	AlertConfiguration `json:",inline" yaml:",inline" xorm:"extends"`
	LastApplied        int64 `xorm:"last_applied"`
}
```

That way it's clear their model is coupled and you won't need `HistoricConfigFromAlertConfig` to convert.